From 17ff0ec0d14305e91a9ad2f503a8aacd0a1c96df Mon Sep 17 00:00:00 2001
From: Matt Stark <msta@google.com>
Date: Fri, 30 Dec 2022 10:13:16 +1100
Subject: [PATCH] Always use the provided bazel if there is one.

Previously, if you passed bazel="//path/to:bazelisk", it would use
tools/bazel when running "bazel info", but the system bazel when
actually running bazel to do dependency analysis.

This is problematic, for example, if you have .bazelversion containing
"6.0.0", and bazel 5.* installed on your system, as it will just
complain that you have the wrong version of bazel.
---
 crate_universe/private/crates_vendor.bzl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/crate_universe/private/crates_vendor.bzl b/crate_universe/private/crates_vendor.bzl
index 5a96b9bc..07df6a2f 100644
--- a/crate_universe/private/crates_vendor.bzl
+++ b/crate_universe/private/crates_vendor.bzl
@@ -227,7 +227,9 @@ def _crates_vendor_impl(ctx):
 
     # Optionally include an explicit `bazel` path
     if ctx.attr.bazel:
-        args.extend(["--bazel", _runfiles_path(ctx.executable.bazel, is_windows)])
+        bazel_path = _runfiles_path(ctx.executable.bazel, is_windows)
+        args.extend(["--bazel", bazel_path])
+        environ["BAZEL_REAL"] = bazel_path
         cargo_bazel_runfiles.append(ctx.executable.bazel)
 
     # Determine platform specific settings
-- 
2.39.0.314.g84b9a713c41-goog

